pkg/graph/resolvers/system_intake_actions.go

```diff
@@ -236,3 +236,222 @@ func CreateSystemIntakeActionRequestEdits(
+
+	// update other fields
+	intake.RejectionReason = null.StringFrom(input.Reason)
+	intake.DecisionNextSteps = null.StringFrom(input.NextSteps)
+	intake.TRBFollowUpRecommendation = &input.TrbFollowUp
+
+	updatedTime := time.Now()
+	intake.UpdatedAt = &updatedTime
+
+	// All the different database calls aren't in a single atomic transaction;
+	// in the case of a system failure, some data from the action might be saved, but not all.
+	// As of this function's initial implementation, we're accepting that risk.
+	// If we create a general way to wrap several store methods calls in a transaction later, we can use that.
+
+	errGroup := new(errgroup.Group)
+	var updatedIntake *models.SystemIntake // declare this outside the function we pass to errGroup.Go() so we can return it
+
+	// save intake
+	errGroup.Go(func() error {
+		var errUpdateIntake error // declare this separately because if we use := on next line, compiler thinks we're declaring a new updatedIntake variable as well
+		updatedIntake, errUpdateIntake = store.UpdateSystemIntake(ctx, intake)
+		if errUpdateIntake != nil {
+			return errUpdateIntake
+		}
+
+		return nil
+	})
+
+	// save action (including additional info for email, if any)
+	errGroup.Go(func() error {
+		action := models.Action{
+			IntakeID:       &input.SystemIntakeID,
+			ActionType:     models.ActionTypeREJECT,
+			ActorName:      adminUserInfo.CommonName,
+			ActorEmail:     adminUserInfo.Email,
+			ActorEUAUserID: adminEUAID,
+			Step:           &intake.Step,
+		}
+		if input.AdditionalInfo != nil {
+			action.Feedback = null.StringFromPtr(input.AdditionalInfo)
+		}
+
+		_, errCreatingAction := store.CreateAction(ctx, &action)
+		if errCreatingAction != nil {
+			return errCreatingAction
+		}
+
+		return nil
+	})
+
+	// save admin note
+	if input.AdminNote != nil {
+		errGroup.Go(func() error {
+			adminNote := &models.SystemIntakeNote{
+				SystemIntakeID: input.SystemIntakeID,
+				AuthorEUAID:    adminEUAID,
+				AuthorName:     null.StringFrom(adminUserInfo.CommonName),
+				Content:        null.StringFromPtr(input.AdminNote),
+			}
+
+			_, errCreateNote := store.CreateSystemIntakeNote(ctx, adminNote)
+			if errCreateNote != nil {
+				return errCreateNote
+			}
+
+			return nil
+		})
+	}
+
+	if err := errGroup.Wait(); err != nil {
+		return nil, err
+	}
+
+	return updatedIntake, nil
+}
+
+// IssueLCID handles a Not Approved by GRB action on an intake as part of Admin Actions v2
+func IssueLCID(
+	ctx context.Context,
+	store *storage.Store,
+	fetchUserInfo func(context.Context, string) (*models.UserInfo, error),
+	input model.SystemIntakeIssueLCIDInput,
+) (*models.SystemIntake, error) {
+	adminEUAID := appcontext.Principal(ctx).ID()
+
+	adminUserInfo, err := fetchUserInfo(ctx, adminEUAID)
+	if err != nil {
+		return nil, err
+	}
+
+	intake, err := store.FetchSystemIntakeByID(ctx, input.SystemIntakeID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = lcidactions.IsIssueLCIDValid(intake)
```

Comment:
It's a nitpick, but I think this would read better if the helper method was something like `IsIntakeValidToIssueLCID`. The current same makes it sound like you're validating the LCID itself which happens below in the `GenerateNewLCID` method if I'm reading correctly.

Reply:
I was thinking along the same lines when writing this, wasn't entirely sure which way to go. Since you pointed it out as well, I'll change the name.

FYI, we don't really validate the LCID itself - if there's an LCID provided in the input, we stuff it in the database, no matter what format it's in. There's at least two possible formats an LCID can be in (I can look up details later), I'm not quite sure what the full logic would be for validating them. I don't think we need to do anything about this right now, but we might want to keep it in mind when we're doing the LCID Management work in v3.

Reply:
Renamed in https://github.com/CMSgov/easi-app/pull/2155/commits/901f8286a348b182a96ffc93b777f7288695ba35.